### ISSUE TITLE
Augment Schemas Processor bug with Items

### ISSUE THREAD

**Issue by Gabb1995** [zircote/swagger-php#890]
```php
class AugmentSchemas
{
    public function __invoke(Analysis $analysis)
    {
        $schemas = $analysis->getAnnotationsOfType(Schema::class);
        // Use the class names for @OA\Schema()
        foreach ($schemas as $schema) {
            if ($schema->schema === UNDEFINED) {
                if ($schema->_context->is('class')) {
                    $schema->schema = $schema->_context->class;
                } elseif ($schema->_context->is('interface')) {
                    $schema->schema = $schema->_context->interface;
                } elseif ($schema->_context->is('trait')) {
                    $schema->schema = $schema->_context->trait;
                }
            }
        }
```

This part of the AugmentSchemas processor also changes the Items class, as the items class extends Schema and has schema property set to undefined.

This invalidates the Items class as it should not have schema property set and fails the validation with swagger-cli.

**Comment by DerManoMann** [comment]
Would you be able to provide a small annotation example to replicate the wrong behaviour? So much easier than words :) (although I think I roughly know what you mean...)

**Comment by DerManoMann** [comment]
Alternatively to my previous question could you please test the attached PR and provide some feedback?

I'd be still interested to learn in what way the `Items` instances got changed by this?

**Comment by Gabb1995** [comment]
Well basically, since Items class was extending Schema class, and Schema class has a property "schema" which by default is set to UNDEFINED, this Processor was changing this "schema" property inside Items classes to the 'context class' name, in our case.

The issue with this was that the swagger-cli validator according to the swagger spec/schema the Items object does not support having "schema" being set. Seems as long as the processor does not amend the Items instances then it should be fine as I saw in the pr.

Unfortunately we decided not to use the swagger-cli validator anymore as it was causing too much changes so I would not be able to test it.
